src/org/zaproxy/zap/extension/websocket/WebSocketSenderScriptListener.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * 
+ */
+package org.zaproxy.zap.extension.websocket;
+
+import java.util.List;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy.Initiator;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy.State;
+
+/**
+ * @author Juha Kivekas
+ *
+ */
+public class WebSocketSenderScriptListener implements WebSocketSenderListener {
+	
+	private ExtensionScript extensionScript;
+
+	public WebSocketSenderScriptListener(){
```

Comment:
Could require the `extensionScript` (we already have it when creating the listener).

Reply:
What would this do if the extension is not available? Log something, throw an exception?

Reply:
Nothing, assume that it's always available (it is, otherwise this is not called).

Reply:
With requiring I was suggesting having the `ExtensionScript` as a parameter of the constructor.